File: api/src/endpoint/question/update_comment.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::{endpoint::Endpoint, resource::question::CommentResource};

pub struct UpdateComment;

impl Endpoint for UpdateComment {
    const PATH: &'static str = "/repository/{owner}/{repo}/question/{number}/comment/{comment_id}";
    const METHOD: http::Method = http::Method::PATCH;

    type Request = UpdateCommentRequest;
    type Response = UpdateCommentResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct UpdateCommentRequest {
    pub body: String,
}

pub type UpdateCommentResponse = CommentResource;

File: gitdot-api-ts/src/endpoint/question/update-comment.ts
import { z } from "zod";

import { CommentResource } from "../../resource";

export const UpdateCommentRequest = z.object({
  body: z.string(),
});
export type UpdateCommentRequest = z.infer<typeof UpdateCommentRequest>;

export const UpdateComment = {
  path: "/repository/{owner}/{repo}/question/{number}/comment/{comment_id}",
  method: "PATCH",
  request: UpdateCommentRequest,
  response: CommentResource,
} as const;
export type UpdateComment = typeof UpdateComment;
